### PR TITLE
#96: Store real dates in DB for student birthdays

### PR DESCRIPTION
**Pull request by chanlawrencet**
To test:
- run python3 resetDatabase.py to prime the database
- run all services
- open tab, navigate to "add student"
- add a student
- see robo3t, collection "student"
- each collection's DOB should have a Datetime object with a zero-ed out time

### CODE REVIEW

File: sfja/src/components/AddStudent/Input.js [hunk 1]
```diff
@@ -17,12 +17,13 @@ const textWidth = {
...
 // eslint-disable-next-line require-jsdoc
 function blankStateExceptSubmitTime(submitTime) {
+  const today = new Date();
   return {
     submitTime: submitTime,
     firstNameStudent: '',
     middleNameStudent: '',
     lastNameStudent: '',
-    dob: new Date(),
+    dob: today.toLocaleDateString(),
```
Comment (chanlawrencet) [inline sfja/src/components/AddStudent/Input.js]: Maybe I should consolidate this to `new Date().toLocaleDateString()`

Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: @jchin01 thoughts?

File: sfja/src/components/AddStudent/Input.js [hunk 2]
```diff
@@ -158,7 +159,7 @@ class Input extends React.Component {
               label="Date picker inline"
               value={dob}
               onChange={(ev) => {
-                this.setState({dob: ev});
+                this.setState({dob: ev.toLocaleDateString()});
```
Comment (chanlawrencet) [inline sfja/src/components/AddStudent/Input.js]: There will be an issue here! Can anyone spot it ;)

File: sfja/src/components/AddStudent/Input.js (hunk 1 above)
Comment (jchin01) [reply sfja/src/components/AddStudent/Input.js]: I think you should consolidate to new Date().toLocaleDateString()

Comment (jchin01) [reply sfja/src/components/AddStudent/Input.js]: I was trying to figure out how to suggest it in code but im not too sure how

Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: All boxes in Github are in Markdown
https://guides.github.com/features/mastering-markdown/

To write a single line of code, surround the code with a tick (`).
To write multiple lines of code, surround the code with three ticks, newline, code, then three more ticks.

Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: I'll put a new commit up!

File: sfja/src/components/AddStudent/Input.js (hunk 2 above)
Comment (jchin01) [reply sfja/src/components/AddStudent/Input.js]: is ev a valid date object?

Comment (jchin01) [reply sfja/src/components/AddStudent/Input.js]: is ev a valid date object?

File: sfja/src/components/AddStudent/Input.js (hunk 1 above)
Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: Fixed with 2c730fa! Resolving.

File: sfja/src/components/AddStudent/Input.js (hunk 2 above)
Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: It usually is. Did you test the branch? It should have no issues in the expected case.

File: sfja/src/components/AddStudent/Input.js [hunk 3]
```diff
@@ -158,7 +158,7 @@ class Input extends React.Component {
...
```
Comment (jchin01) [inline sfja/src/components/AddStudent/Input.js]: Please add a null check to ev

Comment (chanlawrencet) [reply sfja/src/components/AddStudent/Input.js]: should be done with 3ed359f

**Comment by Antranduc** [comment]
Could add functionality for middle name to default to NMN when field is blank instead of making it a required field, but can add at a later date.

**Pull request merged**